pkg/sentry/socket/unix/transport/connectioned.go

```diff
@@ -220,6 +220,9 @@ func (e *connectionedEndpoint) Close() {
 	case e.Connected():
 		e.connected.CloseSend()
 		e.receiver.CloseRecv()
+		if e.receiver.RecvQueuedSize() > 0 {
+			e.connected.SetError(syserr.ErrConnectionReset)
```

Comment:
Rather than storing the data in the queue, maybe it would be better to leave the data in the queue and check for it when attempting to read?

Reply:
If we don't check and flag in Close(), in the read side, we cannot tell if this peer is just shutdown (does not return ECONNRESET) or closed (return ECONNRESET). BTW, I should mention this in commit log, and add a note here.

Reply:
Can we maybe store that rather than the error itself?

Reply:
Yep, besides the orginal state of "closed", we can add a state of "released". Will fix.